compiler/damlc/lib/DA/Cli/Damlc/Test.hs

```diff
@@ -72,25 +77,71 @@ testRun h inFiles lfVersion coverage color mbJUnitOutput  = do
     deps <- runActionSync h $ mapM getDependencies inFiles
     let files = nubOrd $ concat $ inFiles : catMaybes deps
 
+    -- get all external dependencies
+    extPkgs <- fmap (nubSortOn LF.extPackageId . concat) $ runActionSync h $
+      Shake.forP files $ \file -> getExternalPackages file
+    let extDalfs =
+                [ (Just pId, dalf)
+                | pkg <- extPkgs
+                , let dalfs = NM.elems $ LF.packageModules $ LF.extPackagePkg pkg
+                , let pId = LF.extPackageId pkg
+                , dalf <- dalfs
+                ]
+
     results <- runActionSync h $ do
-        dalfs <- Shake.forP files dalfForScenario
         Shake.forP files $ \file -> do
+            dalf <- dalfForScenario file
             mbScenarioResults <- runScenarios file
             mbScriptResults <- runScripts file
-            results <- case liftM2 (++) mbScenarioResults mbScriptResults of
-                Nothing -> failedTestOutput h file
-                Just scenarioResults -> do
-                    -- failures are printed out through diagnostics, so just print the sucesses
-                    let results' = [(v, r) | (v, Right r) <- scenarioResults]
-                    liftIO $ printScenarioResults results' color
-                    liftIO $ printTestCoverage coverage dalfs scenarioResults
-                    let f = either (Just . T.pack . DA.Pretty.renderPlainOneLine . prettyErr lfVersion) (const Nothing)
-                    pure $ map (second f) scenarioResults
-            pure (file, results)
+            let mbResults = liftM2 (++) mbScenarioResults mbScriptResults
+            forM_ mbResults $ \scenarioResults -> do
+                  -- failures are printed out through diagnostics, so just print the successes
+                  let results' = [(v, r) | (v, Right r) <- scenarioResults]
+                  liftIO $ printScenarioResults results' color
+            return (file, dalf, mbResults)
+
+    extResults <- runActionSync h $ do
+        if runAllTests
+        then do
+            projPathM <- liftIO getProjectPath
+            case projPathM of
+                Nothing -> do
+                    liftIO $
+                        putStrLn "Warning: Project root not found, not testing package dependencies"
```

Comment:
why do we need a project root for this? We already have `extPkgs` here.

Reply:
changed it in the meantime. we need a file to create contexts for the scenario service. I changed it to use one of the given input files (if any).